pkg/resource/deploy/source_eval.go

```diff
@@ -419,6 +422,12 @@ func (d *defaultProviders) handleRequest(req providers.ProviderRequest) (provide
 		return ref, nil
 	}
 
+	if d.explicitDefaultProviders[req.Package().Name().String()] {
+		logging.V(9).Infof("package %s has an explicit default provider.  Denying creation of an implicit one",
```

Comment:
I wouldn't have thought we'd ever get here? If a default already exists shouldn't we just be picking up that default, not triggering a registerDefaultProviderEvent?

Reply:
Yeah, I think this made more sense in the previous iteration without the callback based approach, I don't think this is necessary anymore.